packages/plugins/post-multipart-request-parser/src/index.ts

```diff
@@ -0,0 +1,38 @@
+import type { GraphQLParams, Plugin } from '@graphql-yoga/common'
+import { dset } from 'dset'
+
+async function POSTMultipartRequestParser(
+  request: Request,
+): Promise<GraphQLParams> {
+  const requestBody = await request.formData()
+  const operationsStr = requestBody.get('operations')?.toString() || '{}'
+  const operations = JSON.parse(operationsStr)
+
+  const mapStr = requestBody.get('map')?.toString() || '{}'
+  const map = JSON.parse(mapStr)
+  for (const fileIndex in map) {
+    const file = requestBody.get(fileIndex)
+    const [path] = map[fileIndex]
+    dset(operations, path, file)
+  }
+
+  return {
+    operationName: operations.operationName,
+    query: operations.query,
+    variables: operations.variables,
+    extensions: operations.extensions,
+  }
+}
+
+export function usePOSTMultipartRequestParser(): Plugin {
+  return {
+    onRequestParse({ request, setRequestParser }) {
+      if (
+        request.method === 'POST' &&
+        request.headers.get('content-type')?.startsWith('multipart/form-data')
+      ) {
```

Comment:
I am not sure whether I want the `is` logic hard-coded within the plugin.

Ideally the logic building block is also usable outside of the yoga context 🤔.

I liked the previous construct with an `is` and `parse` function.

Also, I am unsure whether all the request parsers should be plugins, providing either one is mandatory - otherwise requests will fail.

Reply:
I see the point of chaining pattern but here we have the same problem we had before in GraphQL Tools loaders. `is`, `canLoad` or that kind of methods might not work if the following "loaders","parsers" has more specific condition.
- POSTRequestParser.is returns true if method is POST
- POSTMultipartParser.is returns true if method is POST and content-type is multipart

For sure we can make POSTRequestParser more specific by adding content-type there but this will be always problematic to stop the iteration if `is` returns true. That's why I wanted to use Envelop's plugins pattern here not only experiment Yoga's new plugin system idea but also to have better configurable "request pipeline".